react/components/RenderProvider.tsx

```diff
@@ -168,6 +168,19 @@ class RenderProvider extends Component<Props, RenderProviderState> {
     }
   }
 
+  public getCustomMessages = (locale: string) => {
+    const {components} = this.state
+    const componentsArray = Object.keys(components)
+
+    const customMessages = componentsArray
+          .map(getImplementation)
+          .filter(component => component && component.getCustomMessages)
+          .map(component => component.getCustomMessages!(locale))
```

Comment:
Why check for `!` if it's being checked on the previous step?

Reply:
Typescript has no way to know `component.getCustomMessages` is not undefined at that point.